app/src/main/java/com/github/onlynotesswent/ui/overview/editnote/Comments.kt

```diff
@@ -0,0 +1,206 @@
+              modifier =
+                  Modifier.fillMaxSize()
+                      .padding(16.dp)
+                      .padding(paddingValues)
+                      .testTag("commentsColumn")
+                      .verticalScroll(rememberScrollState()),
+              verticalArrangement = Arrangement.spacedBy(8.dp),
+              horizontalAlignment = Alignment.CenterHorizontally) {
+                CommentsSection(updatedComments, { updatedComments = it })
+              }
+        }
+      }
+}
+
+/**
+ * Displays a button that adds a new comment to the note. When clicked, the button adds a new
+ * comment to the note in the ViewModel and updates the note's comments and date.
+ *
+ * @param currentUser The current user.
+ * @param updatedComments The updated collection of comments for the note.
+ * @param onCommentsChange The callback function to update the comments collection.
+ */
+@Composable
+fun AddCommentButton(
+    currentUser: User?,
+    note: Note?,
+    updatedComments: Note.CommentCollection,
+    onCommentsChange: (Note.CommentCollection) -> Unit,
+) {
+  if (currentUser != null && note != null) {
+    FloatingActionButton(
+        modifier = Modifier.testTag("addCommentButton"),
+        onClick = {
+          onCommentsChange(
+              updatedComments.addCommentByUser(currentUser.uid, currentUser.userName, ""))
+        },
+        containerColor = MaterialTheme.colorScheme.primary,
+        contentColor = MaterialTheme.colorScheme.onPrimary,
+    ) {
+      Icon(imageVector = Icons.Default.Add, contentDescription = "Add Comment")
+    }
+  }
+}
+
+/**
+ * Displays the comments section of the edit note screen. The section includes a list of comments
+ * with text fields for editing each comment, and buttons for deleting comments and adding new
+ * comments.
+ *
+ * @param updatedComments The collection of comments to be displayed and edited.
+ * @param onCommentsChange The callback function to update the comments collection.
+ */
+@Composable
+fun CommentsSection(
+    updatedComments: Note.CommentCollection,
+    onCommentsChange: (Note.CommentCollection) -> Unit,
+) {
+  if (updatedComments.commentsList.isEmpty()) {
+    Text(
+        text = "No comments yet. Add a comment to start the discussion.",
+        color = Color.Gray,
+        modifier = Modifier.padding(8.dp).testTag("noCommentsText"))
+  } else {
+    updatedComments.commentsList.forEachIndexed { _, comment ->
+      Row(
+          modifier = Modifier.fillMaxWidth().padding(vertical = 4.dp),
+          verticalAlignment = Alignment.CenterVertically, // Align items vertically
+          horizontalArrangement = Arrangement.SpaceBetween // Distribute space
+          ) {
+            // Comment TextField
+            OutlinedTextField(
+                value = comment.content,
+                onValueChange = {
+                  val updatedCollection =
+                      updatedComments.editCommentByCommentId(comment.commentId, it)
```

Comment:
Yes valid point. Instead of modifying it here I modified line 69-72 and 80-83 so that the empty comments don't get saved in the database:
` val commentsNotEmpty = updatedComments.commentsList.filter { it.content.isNotEmpty() }
              noteViewModel.updateNote(
                  note!!.copy(comments = Note.CommentCollection(commentsNotEmpty)),
                  currentUser!!.uid)`